include/nbl/video/IGPUCommandBuffer.h

```diff
@@ -227,6 +227,24 @@ class NBL_API IGPUCommandBuffer :
             for (uint32_t i = _first + 1u; i < IGPUPipelineLayout::DESCRIPTOR_SET_COUNT; ++i)
                 _destPplnLayouts[i] = nullptr;
     }
+
+    uint32_t m_resetCount = 0;
+
+    virtual bool resetCommon()
+    {
+        m_resetCount = m_cmdpool->getResetCounter().load();
```

Comment:
no, you want to set the last-checked-stamp upon a `begin`, not before (you're only interested in transitions _into_ INITIAL or INVALID caused by pool reset)

Reply:
If I have 4 cmdbuffers A,B,C,D and only set the stamp when I reset them manually, then I get a lot of false positives when calling `checkForCommandPoolReset`

Reply:
As things are now, I think `m_resetCount = m_cmdpool->getResetCounter()` should only ever happen in `IGPUCommandBuffer::begin` and perhaps in `IGPUCommandBuffer::reset`?
If we do it in `checkForCommandPoolReset` (according to #373) it will end up getting done in every command buffer method since every command buffer method should call `checkForCommandPoolReset` and if that happens, something like the following could create problems:

```C++
cmdbuf->begin();
cmdpool->reset();
cmdbuf->beginRenderPass(); // detects command pool reset and fails, BUT updates the command buffer last-checked-stamp
cmdbuf->draw(); // doesn't detect command pool reset and tries to do work, and crashes
..
```

We could also solve this by doing state validation in the command buffer methods, something like:
```C++
if (m_state != ES_RECORDING)
    return false;
```

Reply:
I think I did what you asked for here, but it would be nice if you still answer the question above.

Reply:
ok so we definitely need to set the timestamp in begin()
you shouldn't update the timestamp after a "successful" checkForParentPoolReset, to make sure it always returns true if there was a reset since last begin()

> We could also solve this by doing state validation in the command buffer methods, something like:

great idea, we could roll it into a single function

```cpp
bool checkStateBeforeRecording()
{
    if (m_state != ES_RECORDING)
    {
        m_logger.log("Failed to record into command buffer: not in RECORDING state.", system::ILogger::ELL_ERROR);
        return false;
    }
    if (checkForParentPoolReset())
    {
        m_logger.log("Failed to record into command buffer: pool was reset since the recording begin() call.", system::ILogger::ELL_ERROR);
        return false;
    }
    return true;
}
```

and use that as the prelude to any IGPUCommandBuffer method (including `end()`)

Reply:
Resolved.